idmapping-rest/src/main/java/org/uniprot/api/idmapping/controller/IdMappingDownloadController.java

```diff
@@ -0,0 +1,136 @@
+            @Valid @ModelAttribute IdMappingDownloadRequestImpl request) {
+        validateRequest(request);
+        String jobId = this.messageService.sendMessage(request);
+        return ResponseEntity.ok(new JobSubmitResponse(jobId));
+    }
+
+    @GetMapping(
+            value = "/status/{jobId}",
+            produces = {APPLICATION_JSON_VALUE})
+    @Operation(
+            summary = "Get the status of a idmapping download job.",
+            responses = {
+                @ApiResponse(
+                        content = {
+                            @Content(
+                                    mediaType = APPLICATION_JSON_VALUE,
+                                    schema = @Schema(implementation = JobStatus.class))
+                        })
+            })
+    public ResponseEntity<JobStatusResponse> getJobStatus(@PathVariable String jobId) {
+        Optional<DownloadJob> optJob = jobRepository.findById(jobId);
+        DownloadJob job = getAsyncDownloadJob(optJob, jobId);
+        return getAsyncDownloadStatus(job);
+    }
+
+    @GetMapping(
+            value = "/details/{jobId}",
+            produces = {APPLICATION_JSON_VALUE})
+    public ResponseEntity<DownloadJobDetailResponse> getDetails(
+            @PathVariable String jobId, HttpServletRequest servletRequest) {
+
+        Optional<DownloadJob> optJob = this.jobRepository.findById(jobId);
+        DownloadJob job = getAsyncDownloadJob(optJob, jobId);
+
+        DownloadJobDetailResponse detailResponse = new DownloadJobDetailResponse();
+        detailResponse.setFields(job.getFields());
+        detailResponse.setFormat(job.getFormat());
+        if (JobStatus.FINISHED == job.getStatus()) {
+            detailResponse.setRedirectURL(
+                    constructDownloadRedirectUrl(
+                            job.getResultFile(), servletRequest.getRequestURL().toString()));
+        } else if (JobStatus.ERROR == job.getStatus()) {
+            ProblemPair error =
+                    new ProblemPair(PredefinedAPIStatus.SERVER_ERROR.getCode(), job.getError());
+            detailResponse.setErrors(List.of(error));
+        }
+
+        return ResponseEntity.ok(detailResponse);
+    }
+
+    private void validateRequest(IdMappingDownloadRequestImpl request) {
+        String jobId = request.getJobId();
+        IdMappingJob idMappingJob = idMappingJobCacheService.getJobAsResource(jobId);
+        if (idMappingJob.getJobStatus() != FINISHED) {
+            throw new InvalidRequestException(
+                    "IdMapping Job must be finished, before we start to download it.");
+        }
+        String type = idMappingJob.getIdMappingRequest().getTo();
+        IdMappingDownloadRequestValidatorFactory validatorFactory =
+                new IdMappingDownloadRequestValidatorFactory();
+        IdMappingDownloadRequestValidator validator = validatorFactory.create(type);
+        validator.validate(request);
+    }
```

Comment:
We can not, because in the extended classes we have what is different (field names and formats). 
UniRef/UniParc and UniProtKB has differents formats and field names.